src/Rector/Deprecation/Base/FunctionToImmutableConfigBase.php

```diff
@@ -0,0 +1,47 @@
+<?php declare(strict_types=1);
+
+namespace DrupalRector\Rector\Deprecation\Base;
+
+use PhpParser\Node;
+use Rector\Core\Rector\AbstractRector;
+
+
+/**
+ * Replace deprecated function calls with a wrapped config.factory service call to get an immutable configuration.
+ *
```

Comment:
Can you add a comment describing the scope of this base class like the example below?

```
What is covered:
 - Static replacement
Improvement opportunities
 - Dependency injection
```